File: frontend/src/types/notification.ts
export interface AppNotification {
  id: string;
  type: string;
  title: string;
  body: string | null;
  lead_id: string | null;
  read_at: string | null;
  created_at: string;
  reference_type: string;
  reference_id: string;
}

export interface NotificationListResponse {
  items: AppNotification[];
  total: number;
}

File: app/schemas/notification.py
from uuid import UUID
from datetime import datetime
from typing import Optional, List
from pydantic import BaseModel, ConfigDict


class NotificationResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: UUID
    type: str
    title: str
    body: Optional[str] = None
    lead_id: Optional[UUID] = None
    read_at: Optional[datetime] = None
    created_at: datetime
    reference_type: str
    reference_id: UUID


class NotificationListResponse(BaseModel):
    items: List[NotificationResponse]
    total: int


class UnreadCountResponse(BaseModel):
    count: int
